WordPress/Classes/Utility/Spotlight/SearchManager.swift

```diff
@@ -252,37 +252,45 @@ fileprivate extension SearchManager {
                    blogID: NSNumber,
                    onSuccess: @escaping (_ post: AbstractPost) -> Void,
                    onFailure: @escaping () -> Void) {
-        let context = ContextManager.sharedInstance().mainContext
+        let coreDataStack = ContextManager.shared
 
-        guard let blog = Blog.lookup(withID: blogID, in: context) else {
+        guard let blog = Blog.lookup(withID: blogID, in: coreDataStack.mainContext) else {
                 onFailure()
                 return
         }
 
-        let postService = PostService(managedObjectContext: context)
-        postService.getPostWithID(postID, for: blog, success: { apost in
-            onSuccess(apost)
-        }, failure: { error in
-            onFailure()
-        })
+        let postRepository = PostRepository(coreDataStack: coreDataStack)
+        Task { @MainActor in
+            do {
+                let postObjectID = try await postRepository.getPost(withID: postID, from: .ofSaved(blog))
+                let post = try postObjectID.existingObject(in: coreDataStack.mainContext)
```

Comment:
How do you feel of making this a functionality in `NSManagedContext` instead?

I didn't occur to me when looking at the implementation, but now that I see it in action it looks a bit odd to extract a model instance from an object representing its identifier.

The alternative I have in mind would be something like:

```swift
let post = try coreDataStack.mainContext.existingObject(for: postObjectID)
```

I did a quick experiment locally. This diff had the tests passing

```diff
diff --git a/WordPress/Classes/Utility/CoreDataHelper.swift b/WordPress/Classes/Utility/CoreDataHelper.swift
index d2dfb30b24..7eba4e58ce 100644
--- a/WordPress/Classes/Utility/CoreDataHelper.swift
+++ b/WordPress/Classes/Utility/CoreDataHelper.swift
@@ -511,3 +511,25 @@ struct CoreDataObjectIdentifier<Model: NSManagedObject> {
         }
     }
 }
+
+extension NSManagedObjectContext {
+
+    func existingObject<Model: NSManagedObject>(with identifier: CoreDataObjectIdentifier<Model>) throws -> Model {
+        do {
+            var result: Result<Model, Error>!
+
+            // Catch an Objective-C `NSInvalidArgumentException` exception from `existingObject(with:)`.
+            // See https://github.com/wordpress-mobile/WordPress-iOS/issues/20630
+            try WPException.objcTry {
+                result = Result {
+                    let object = try self.existingObject(with: identifier.objectID)
+                    return object as! Model
+                }
+            }
+
+            return try result.get()
+        } catch {
+            throw error
+        }
+    }
+}
diff --git a/WordPress/WordPressTest/PostRepositoryTests.swift b/WordPress/WordPressTest/PostRepositoryTests.swift
index 8d4c6d1770..99043ca46f 100644
--- a/WordPress/WordPressTest/PostRepositoryTests.swift
+++ b/WordPress/WordPressTest/PostRepositoryTests.swift
@@ -50,6 +50,12 @@ class PostRepositoryTests: CoreDataTestCase {
         XCTAssertTrue(isPage)
         XCTAssertEqual(title, "Post: Test")
         XCTAssertEqual(content, "This is a test post")
+        let isPage2 = try await contextManager.performQuery { try $0.existingObject(with: postID) is Page }
+        let title2 = try await contextManager.performQuery { try $0.existingObject(with: postID).postTitle }
+        let content2 = try await contextManager.performQuery { try $0.existingObject(with: postID).content }
+        XCTAssertTrue(isPage2)
+        XCTAssertEqual(title2, "Post: Test")
+        XCTAssertEqual(content2, "This is a test post")
     }
 
 }
```

<img width="1477" alt="image" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/00b15a42-46af-4fff-a256-f25061c87945">

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/21163/commits/321f6c9965112ca7f64e3c9c8d038254fc116e4e